Event card markup and styles

helenb opened the pull request:
[Ticket URL](https://national-archives.atlassian.net/wiki/spaces/DSWhatson/pages/143294466/Event+card+specification)

## About these changes
This adds the markup and styling for the event card. It adds one event card to the 'whats on listing' page with hard coded values. Note that the dividing lines in the design will come with the page layout. Likewise, the full-width image at mobile will be part of the 'whats on listing' page layout

## Details to be aware of
In storybook, the meta information is assumed to be supplied as a list. At the moment, I can't add an event page on my local build without an error, but it appears that the date and venue information are completely separate fields, and unlikely to come through as a list unless the context is specifically updated. It may be that we want to take this route but that felt like a wider team discussion - for now I have added them as separate fields.

Also to discuss:
- the design shows just one date, not a range
- I can't see a price / cost field in the event page

## How to check these changes
- Check out the `feature/event-card-fe` branch
- In your local build, create a "what's on" listing page and preview it

## Screenshots
<details>
<summary>
Click to view
</summary>

Desktop

![Screenshot 2023-10-04 at 12 22 39](https://github.com/nationalarchives/ds-wagtail/assets/771869/9e3770f9-97c3-4754-9596-e5d82ac2fcc6)

Mobile

![Screenshot 2023-10-04 at 12 23 08](https://github.com/nationalarchives/ds-wagtail/assets/771869/59dfa0ef-68af-470d-b831-0759ad643642)

</details>

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer
- [ ] Checked PR title starts with ticket number as per project conventions to help us keep track of changes
- [x] Ensured that PR includes only commits relevant to the ticket
- [x] Waited for all CI jobs to pass before requesting a review
- [ ] Added/updated tests and documentation where relevant

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

helenb commented:
@ahosgood I've requested a review from Nick on this but also tagging you if you would like to check it as well.

jamesbiggs commented:
I've hooked up most of the variables we have from the BE. The only ones that are missing is the "sold out" status, the cost, and I need to figure out how to display the label of the `venue_type`, so for now it might show `IN_PERSON` if it's an in person event. Otherwise this all looks good to me! But I shall leave it with AJ/Nick to do a proper review 😃 

Nice work @helenb!

helenb commented:
> I've hooked up most of the variables we have from the BE. The only ones that are missing is the "sold out" status, the cost, and I need to figure out how to display the label of the `venue_type`, so for now it might show `IN_PERSON` if it's an in person event. Otherwise this all looks good to me! But I shall leave it with AJ/Nick to do a proper review 😃
> 
> Nice work @helenb!

Thanks for adding those @jamesbiggs . I have adjusted the image crop and added the correct values for the image width, height and alt to the card include.

Variables still needed from back-end:
- The 'highlight' value (e.g. 'sold out' or 'opening soon')
- The highlight colour - determined by the wording of the highlight
- The cost

@jamesbiggs we also need to update the filters with the correct fields from wagtail, as the form is all hard-coded for now. You may also need to change the names and values of the fields depending on what the back-end is expecting in terms of filtering the events - but that might be one for another ticket.

helenb commented:
Noting that from discussion with @nicklee the colours of 'sold out' and 'opening soon' etc may be changing, but we don't know for certain what to yet, so changing these will be a future enhancement

jamesbiggs commented:
I'll be adding the cost + status soon, after a chat with Lauren etc 😃 And I'll take a look at the forms too 🙌

nicklee commented:
This looks great, interesting to see all the new mixins in use, should make everything more consistent going forward 👍 

As mentioned on slack, from a visual pov, the titles should be black, and there shouldn't be an arrow after them. But other than that, all looks good.

ahosgood commented on templates/components/card.html:
```diff
@@ -60,6 +78,40 @@
     <div class="tna-card__body">
+    {# for now I am splitting these out as this depends on how the data is recieved from the back-end #}
+    {% if date or cost or venue_type %}
+      <ul class="tna-card__meta">
```
Hang on - I've just realised that you are relying on these new styles to be deployed!

Ignore the previous comment 🤦🏼‍♂️

ahosgood commented on templates/components/card.html:
```diff
@@ -26,13 +38,19 @@
       {% endif %}
     {% else %}
       <h{{ headingLevel }} class="tna-card__heading tna-heading tna-heading--{{ headingSize|default:'m' }}">
+      <h{{ headingLevel }} class="tna-card__heading tna-heading tna-heading--{{ headingSize|default:'m' }} {% if highlight %}tna-card__heading--has-highlight{% endif %}" >
```
Is this duplicated opening heading tags?

helenb replied:
oops, good spot - was when I was porting in the tna-frontend styles and making sure everything was in line with the card as it stands
